sockeye/encoder.py

```diff
@@ -1008,19 +1027,44 @@ class TransformerEncoder(Encoder):
     def __init__(self,
                  config: transformer.TransformerConfig,
                  prefix: str = C.TRANSFORMER_ENCODER_PREFIX) -> None:
-        super().__init__(config.dtype)
+        Encoder.__init__(self, dtype=config.dtype)
+        mx.gluon.HybridBlock.__init__(self, prefix=prefix)
         self.config = config
-        self.prefix = prefix
-        self.layers = [transformer.TransformerEncoderBlock(
-            config, prefix="%s%d_" % (prefix, i)) for i in range(config.num_layers)]
-        self.final_process = transformer.TransformerProcessBlock(sequence=config.preprocess_sequence,
-                                                                 dropout=config.dropout_prepost,
-                                                                 prefix="%sfinal_process_" % prefix)
+
+        with self.name_scope():
+            self.layers = mx.gluon.nn.HybridSequential()
+            for i in range(config.num_layers):
+                self.layers.add(transformer.TransformerEncoderBlock(config, prefix="%d_" % i))
+            self.valid_length_mask = transformer.TransformerValidLengthMask(num_heads=self.config.attention_heads,
+                                                                            fold_heads=True,
+                                                                            name="bias")
+            self.final_process = transformer.TransformerProcessBlock(sequence=config.preprocess_sequence,
+                                                                     dropout=config.dropout_prepost,
+                                                                     prefix="final_process_")
+
+    def hybrid_forward(self, F, data, data_length):
+        return self._encode(F, data, data_length)
+
+    def _encode(self, F, data: mx.sym.Symbol, data_length: mx.sym.Symbol) -> mx.sym.Symbol:
+
```

Comment:
also nit: replace the vertical space with a doc string?

Reply:
removed the vertical space :)